src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,291 @@
+        }
+    }
+
+    override fun searchMangaParse(response: Response) = parseSearchList<SearchResponse>(response)
+
+    // Comic details
+    override fun mangaDetailsRequest(manga: SManga) = comicByIDRequest(manga.url.substringAfterLast("/"))
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val res = response.parseAs<Data<ComicByIDResponse>>()
+        val comic = res.data.comic.toSManga()
+        return comic
+    }
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+
+    /**
+     * 解析日期
+     * Parse date
+     */
+    private fun parseDate(dateStr: String): Long {
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+        sdf.timeZone = TimeZone.getTimeZone("UTC")
+        return sdf.parse(dateStr)?.time ?: 0L
+    }
+
+    /**
+     * 解析章節列表
+     * Parse chapter list
+     */
+    private inline fun <reified T : ChaptersResult> parseChapterList(response: Response, comicUrl: String): List<SChapter> {
+        val res = response.parseAs<Data<T>>()
+        val comics = res.data.chapters
+
+        val tChapters = comics.filter { it.type == "chapter" }
+        val tBooks = comics.filter { it.type == "book" }
+
+        val entries = (tChapters + tBooks).map { chapter ->
+            SChapter.create().apply {
+                url = "$comicUrl/chapter/${chapter.id}/page/1"
+                name = when (chapter.type) {
+                    "chapter" -> "第 ${chapter.serial} 話"
+                    "book" -> "第 ${chapter.serial} 卷"
+                    else -> chapter.serial
+                }
+                date_upload = parseDate(chapter.dateCreated)
+                chapter_number = chapter.serial.toFloatOrNull() ?: -1f
+            }
+        }.reversed()
+
+        return entries
+    }
+
+    // Chapter list
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val payload = Payload(
+            operationName = "chapterByComicId",
+            variables = ChapterByComicIdVariables(manga.url.substringAfterLast("/")),
+            query = QUERY_CHAPTER,
+        ).toJsonRequestBody()
+
+        return client.newCall(POST(queryAPIUrl, headers, payload))
+            .asObservableSuccess()
+            .map { response ->
+                parseChapterList<ChaptersResponse>(response, manga.url)
+            }
+    }
+
+    override fun chapterListParse(response: Response) = throw UnsupportedOperationException()
+
+    /**
+     * 檢查 API 是否達到上限
+     * Check if the API has reached its limit.
+     *
+     * (Idk how to throw an exception in reading page)
+     */
+    private fun fetchAPILimit(): Boolean {
+        val payload = Payload("getImageLimit", "", QUERY_API_LIMIT).toJsonRequestBody()
+        val response = client.newCall(POST(queryAPIUrl, headers, payload)).execute()
+        val limit = response.parseAs<APILimitData>().getImageLimit
+        return limit.limit <= limit.usage
+    }
```

Comment:
unused?

Reply:
like comment. how to throw an exception in reading page? i throw in imageRequest but only have "Failed to load image". how to custom this error text?
![image](https://github.com/user-attachments/assets/53f55553-f7b3-4f69-b87a-76970daf4a29)

Reply:
you can't unfortunately. what is the api limit representing? should it be thrown when page list is fetched?

Reply:
> you can't unfortunately. what is the api limit representing? should it be thrown when page list is fetched?

it's comic image request limit.

Reply:
> you can't unfortunately. what is the api limit representing? should it be thrown when page list is fetched?

in early commit. it's in the page list fetched. if this limit is reached while reading. reader just output "Failed to load image"